lib/scholar/cluster/k_means.ex

```diff
@@ -364,8 +365,8 @@ defmodule Scholar.Cluster.KMeans do
   end
 
   deftransformp validate_weights(weights, num_samples, num_runs) do
-    if is_nil(weights) or
-         (is_list(weights) and length(weights) == num_samples) do
+    if is_nil(weights) or (is_list(weights) and length(weights) == num_samples) or
+         (Nx.is_tensor(weights) and Nx.shape(weights) == {num_samples}) do
```

Comment:
I'd assert on the shape after converting from a list, because otherwise your list can be nested and would become an N-dimensional tensor instead